client/composables/useAPI.ts

```diff
@@ -4,11 +4,9 @@ import type { AxiosResponse, RawAxiosRequestConfig } from 'axios'
 
 export default function useAPI<T = any, R = AxiosResponse<T>, D = any>(
   url: string,
-  config?: RawAxiosRequestConfig<D>,
+  config?: RawAxiosRequestConfig<D> & { skipAuthRefresh?: boolean },
   options?: UseAxiosOptions<T>,
 ): StrictUseAxiosReturn<T, R, D> & PromiseLike<StrictUseAxiosReturn<T, R, D>> {
-  const appConfig = useRuntimeConfig()
-  if (!options)
-    return useAxios(url, { ...config, baseURL: appConfig.public.apiBaseUrl })
-  return useAxios(url, { ...config, baseURL: appConfig.public.apiBaseUrl }, options)
+  const axiosInstance = useAxiosInstance()
+  return useAxios(url, config ?? {}, axiosInstance.value, options ?? {})
```

Comment:
So did you manage to fix the weird issue when using Axios without options?

Reply:
I just give an empty object, since it's what is actually done behind the scenes when calling the "overload" without said parameters

Reply:
Did you test using it with no `{ immediate: false }`?

Reply:
I actually think you got a good catch.

I thought that immediate false was the default behavior, since when i dont pass it, it acts so.

But now that you mention it, I don't think that should be the case, and it might be because of what i've done.

I'll have to fix it. (maybe)

Reply:
you were right, i'll revert to how it was before